syntax = "proto3";

package chat;

// Serviço para gerenciar usuários
service UserService {
  rpc RegisterUser (RegisterUserRequest) returns (RegisterUserResponse);
}

// Mensagem para registrar um usuário
message RegisterUserRequest {
  string username = 1;
}

// Resposta para o registro de usuário
message RegisterUserResponse {
  bool success = 1;
  string message = 2;
}

// Serviço para gerenciar mensagens
service MessageService {
  rpc SendMessage (Message) returns (SendMessageResponse);
  rpc ListMessages (ListMessagesRequest) returns (stream Message); // Usaremos streaming para novas mensagens
}

// Mensagem de chat
message Message {
  string sender = 1;
  string content = 2;
  int64 timestamp = 3; // Para ordenação
}

// Resposta para o envio de mensagem
message SendMessageResponse {
  bool success = 1;
  string message = 2;
}

// Mensagem para listar mensagens
message ListMessagesRequest {
  // Pode adicionar filtros ou paginação aqui no futuro, se necessário
}